### PR TITLE
Ensure retrieved files from github can only be ".yaml"

### LINKED ISSUE #124: WingetCreate copies all files.  It should be limited to .YAML
When updating a package, wingetcreate will pull down the previous version (yay!) so that most fields are already filled in.

The problem is that it pulls down more than *.yaml.  Since only YAML files can be submitted, this can cause submission errors.  Please only recreate YAML files.

### PR DESCRIPTION
**Pull request by ryfu-msft**
Fixes #124 

Adds an additional check when retrieving files from GitHub that the extension must be == ".yaml".

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-create/pull/126)

### CODE REVIEW

File: src/WingetCreateCore/Common/GitHub.cs [hunk 1]
```diff
@@ -103,7 +104,7 @@ public async Task<List<string>> GetLatestManifestContentAsync(string packageId)
                 .FirstOrDefault();
 
             var packageContents = (await this.github.Repository.Content.GetAllContents(this.wingetRepoOwner, this.wingetRepo, version))
-                .Where(c => c.Type != ContentType.Dir);
+                .Where(c => c.Type != ContentType.Dir && Path.GetExtension(c.Name) == ".yaml");
```
Comment (palenshus) [inline src/WingetCreateCore/Common/GitHub.cs:107]: should be case-insensitive

**Review by palenshus — CHANGES_REQUESTED** [review]
:clock1:

File: src/WingetCreateCLI/Commands/NewCommand.cs [hunk 2]
```diff
@@ -1,142 +1,142 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license.
-
-namespace Microsoft.WingetCreateCLI.Commands
-{
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.WingetCreateCLI.Commands
```
Comment (palenshus) [inline src/WingetCreateCLI/Commands/NewCommand.cs:4]: Revert this file, only whitespace changed

File: src/WingetCreateCore/Common/GitHub.cs (hunk 1 above)
Comment (ryfu-msft) [reply src/WingetCreateCore/Common/GitHub.cs]: Changed to a case-insensitive string comparison

File: src/WingetCreateCLI/Commands/NewCommand.cs (hunk 2 above)
Comment (ryfu-msft) [reply src/WingetCreateCLI/Commands/NewCommand.cs]: reverted this file

File: src/WingetCreateCore/Common/GitHub.cs (hunk 1 above)
Comment (palenshus) [reply src/WingetCreateCore/Common/GitHub.cs]: We have an extension method for this already, EqualsIC

**Review by palenshus — APPROVED** [review]

**Pull request merged**